x/auth/tests/abci.rs

```diff
@@ -0,0 +1,128 @@
+use auth::{AuthABCIHandler, GenesisState, Keeper};
+use gears::{
+    tendermint::types::time::timestamp::Timestamp,
+    utils::node::{init_node, GenesisSource, MockOptionsFormer},
+};
+use std::str::FromStr;
+
+use gears::{
+    params::{ParamsSubspaceKey, SubspaceParseError},
+    store::StoreKey,
+    types::address::AccAddress,
+    x::module::Module,
+};
+
+#[test]
+/// In this scenario, we test the initialization of the application and execute a few blocks
+fn test_init_and_few_blocks() {
+    let opt: MockOptionsFormer<
+        SubspaceKey,
+        AuthABCIHandler<SpaceKey, SubspaceKey, AuthModules>,
+        GenesisState,
+    > = MockOptionsFormer::new()
+        .abci_handler(AuthABCIHandler::new(Keeper::new(
+            SpaceKey::Auth,
+            SubspaceKey::Auth,
+            AuthModules::FeeCollector,
+        )))
+        .baseapp_sbs_key(SubspaceKey::BaseApp)
+        .genesis(GenesisSource::Genesis(GenesisState::default()));
+
+    let (mut node, _) = init_node(opt);
+
+    let app_hash = node.step(vec![], Timestamp::UNIX_EPOCH);
+    assert_eq!(
+        data_encoding::HEXLOWER.encode(app_hash),
+        "8d3663f81a98bec58a6d3a9f39c38469438bfecd8257dd335c8c047b933b08ad"
+    );
+
+    for _ in 0..100 {
+        node.step(vec![], Timestamp::UNIX_EPOCH);
+    }
+
+    let app_hash = node.step(vec![], Timestamp::UNIX_EPOCH);
+    assert_eq!(
+        data_encoding::HEXLOWER.encode(app_hash),
+        "8d3663f81a98bec58a6d3a9f39c38469438bfecd8257dd335c8c047b933b08ad"
+    );
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AuthModules {
+    FeeCollector,
+}
+
+impl Module for AuthModules {
+    fn get_name(&self) -> String {
+        match self {
+            AuthModules::FeeCollector => "fee_collector".into(),
+        }
+    }
+
+    fn get_address(&self) -> AccAddress {
+        match self {
+            AuthModules::FeeCollector => {
+                AccAddress::from_bech32("cosmos17xpfvakm2amg962yls6f84z3kell8c5lserqta")
+                    .expect("hard coded address is valid")
+            }
+        }
+    }
+
+    fn get_permissions(&self) -> Vec<String> {
+        match self {
+            AuthModules::FeeCollector => vec![],
+        }
+    }
+}
+
+#[derive(strum::EnumIter, Debug, PartialEq, Eq, Hash, Clone)]
+pub enum SpaceKey {
+    Auth,
+    Params,
+}
+
+#[derive(strum::EnumIter, Debug, PartialEq, Eq, Hash, Clone)]
+pub enum SubspaceKey {
```

Comment:
I'm not sure that we need similar enums for each test file

Reply:
I'm unsure how to do this. Maybe new crate with all stuff from `utils` + these enum with all possible xmods?

Reply:
At least we can use one extended enum for tests inside mod. What do you think? 
But it is better to use single enum for all tests because it allows to reduce boilerplate.

Reply:
I want to avoid any connection between xmods, but this creates such boilerplate. Main the issue is that we should create sort of mocks for our traits. Enum is the smallest issue that may be fixed with one large variant for test purposes